torchaudio/models/_wavernn.py

```diff
@@ -1,105 +1,325 @@
+            nn.BatchNorm1d(n_hidden),
             nn.ReLU(inplace=True),
             *ResBlocks,
-            nn.Conv1d(in_channels=hidden_dims, out_channels=output_dims, kernel_size=1)
+            nn.Conv1d(in_channels=n_hidden, out_channels=n_output, kernel_size=1)
         )
 
     def forward(self, x: Tensor) -> Tensor:
-        r"""Pass the input through the _MelResNet layer.
-
+        r"""
         Args:
-            x: the input sequence to the _MelResNet layer (required).
+            x: the input sequence to the _MelResNet layer
 
         Shape:
-            - x: :math:`(N, S, T)`.
-            - output: :math:`(N, P, T - 2 * pad)`.
-        where N is the batch size, S is the number of input sequence,
-        P is the number of output sequence, T is the length of input sequence.
+            - x: :math:`(batch, freq, time)`
+            - output: :math:`(batch, n_output, time - kernel_size + 1)`
         """
 
         return self.melresnet_model(x)
+
+
+class _Stretch2d(nn.Module):
+    r"""Two-dimensional stretch layer.
+
+    Args:
+        x_scale: the scale factor in x axis
+        y_scale: the scale factor in y axis
+
+    Examples::
+        >>> stretch2d = _Stretch2d(x_scale=10, y_scale=10)
+
+        >>> input = torch.rand(10, 1, 100, 512)
+        >>> output = stretch2d(input)
+    """
+
+    def __init__(self,
+                 x_scale: int,
+                 y_scale: int) -> None:
+        super().__init__()
+
+        self.x_scale = x_scale
+        self.y_scale = y_scale
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""
+        Args:
+            x: the input sequence to the _Stretch2d layer
+
+        Shape:
+            - x: :math:`(..., freq, time)`
+            - output: :math:`(..., freq * y_scale, time * x_scale)`
+        """
+
+        return x.repeat_interleave(self.y_scale, 2).repeat_interleave(self.x_scale, 3)
+
+
+class _UpsampleNetwork(nn.Module):
+    r"""Upsample block based on a stack of Conv2d and Strech2d layers.
+
+    Args:
+        upsample_scales: the list of upsample scales
+        n_res_block: the number of ResBlock in stack (default=10)
+        n_freq: the number of bins in a spectrogram (default=128)
+        n_hidden: the number of hidden dimensions (default=128)
+        n_output: the number of output dimensions (default=128)
+        kernel_size: the number of kernel size in the first Conv1d layer (default=5)
+
+    Examples::
+        >>> upsamplenetwork = _UpsampleNetwork(upsample_scales=[4, 4, 16],
+                                               n_res_block=10,
+                                               n_freq=128,
+                                               n_hidden=128,
+                                               n_output=128,
+                                               kernel_size=5)
+        >>> input = torch.rand(10, 128, 512)
+        >>> output = upsamplenetwork(input)
```

Comment:
This example is missing the most important information of what the output shape looks like

Reply:
The output shape is added.